### PR TITLE
feat: add additional input parameters and child module for App Insights AVM module

### LINKED ISSUE #895: [AVM Module Issue]: App Insights Resource module missing parameters
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/insights/component

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

0.1.2

### Description

The module is missing parameters and child modules:
parameters:

- disableIpMasking
- disableLocalAuth
- forceCustomerStorageForProfiler

Child module:
- [linkedStorageAccounts](https://learn.microsoft.com/en-us/azure/templates/microsoft.insights/components/linkedstorageaccounts?pivots=deployment-language-bicep)

I have updated this module in my environment that included all above listed missing components. I will raise a PR later.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by tyconsulting**
## Description

closes #895 

## Pipeline

[avm.res.insights.component](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [x] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by AlexanderSehr** [comment]
Hey @tyconsulting, thanks for your contribution. @krbar will review it once he's back in the office :) 
We're on it.

File: avm/res/insights/component/main.bicep [hunk 1]
```diff
@@ -107,6 +122,14 @@ resource appInsights 'Microsoft.Insights/components@2020-02-02' = {
   }
 }
 
+module linkedStorageAccount 'linkedStorageAccounts/main.bicep' = if (!empty(linkedStorageAccountResourceId)) {
+  name: '${name}-linkedStorageAccount'
```
Comment (krbar) [inline avm/res/insights/component/main.bicep:126]:
```suggestion
  name: '${uniqueString(deployment().name, location)}-appInsights-linkedStorageAccount'
```

**Review by krbar — COMMENTED** [review]
Hello @tyconsulting, nice contribution, thank you! Looks good to me, I just added a small suggestion to align the deployment name of the child module to the other. A re-generation of the ARM file will be needed. Please also re-run the workflow after the change.

**Comment by tyconsulting** [comment]
> Hello @tyconsulting, nice contribution, thank you! Looks good to me, I just added a small suggestion to align the deployment name of the child module to the other. A re-generation of the ARM file will be needed. Please also re-run the workflow after the change.

hi @krbar thanks for reviewing this PR. I have updated it as per your suggestion. Please take another look. here's the result of the latest pipeline run:

[avm.res.insights.component](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)

**Comment by krbar** [comment]
@Azure/avm-core-team-technical-bicep PR is fine from my point of view and can be merged, please approve.

**Review by jtracey93 — APPROVED** [review]
approving as advised by @krbar

**Pull request merged**